api-gateway/src/main/java/com/example/api/gateway/web/controller/GenerateController.java

```diff
@@ -0,0 +1,91 @@
+/***
+<p>
+    Licensed under MIT License Copyright (c) 2025 Raja Kolli.
+</p>
+***/
+
+package com.example.api.gateway.web.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import java.net.URI;
+import java.time.Duration;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+/** Controller that makes sequential service calls with a delay in between. */
+@RestController
+@RequestMapping("/api/generate")
+public class GenerateController {
```

Comment:
@coderabbitai, can you please help me with writing integration test use wiremock for stubbing